components/cas-inbound-authenticator/src/main/java/org/wso2/carbon/identity/sso/cas/request/CASIdentityRequestFactory.java

```diff
@@ -64,19 +64,29 @@ public int getPriority() {
     @Override
     public IdentityRequest.IdentityRequestBuilder create(HttpServletRequest request,
                                                          HttpServletResponse response) throws FrameworkClientException {
-        String serviceProviderUrl = request.getParameter(CASSSOConstants.SERVICE_PROVIDER_ARGUMENT);
-        String ticket = request.getParameter(CASSSOConstants.SERVICE_TICKET_ARGUMENT);
+        String serviceProviderUrl = request.getParameter(CASConstants.CASSSOConstants.SERVICE_PROVIDER_ARGUMENT);
+        String ticket = request.getParameter(CASConstants.CASSSOConstants.SERVICE_TICKET_ARGUMENT);
         CASIdentityRequest.CASIdentityRequestBuilder builder;
         if (!StringUtils.isEmpty(serviceProviderUrl)) {
             if (StringUtils.isEmpty(ticket)) {
-                builder = new CASSpInitRequest.CASSpInitRequestBuilder(request, response);
+                builder = new CASSInitRequest.CASSpInitRequestBuilder(request, response);
             } else {
                 builder = new CASServiceValidateRequest.CASServiceValidateRequestBuilder(request, response);
                 ((CASServiceValidateRequest.CASServiceValidateRequestBuilder) builder).setLocale(request);
             }
         } else {
             throw CAS2ClientException.error("Invalid request message or invalid service url");
         }
+
+        Enumeration attributeNames = request.getAttributeNames();
```

Comment:
We remove non serialised attributes for now. Once they fix the issue in framework level, we need to remove this part from the authenticator.